src/autopas/containers/verletListsCellBased/verletListsCells/PairwiseVerletNeighborList.h

```diff
@@ -0,0 +1,140 @@
+ * and generates a neighbor list for each particle from cell1, which consists of its (potential) partners from cell2.
+ * @tparam Particle Type of particle to be used for this neighbor list.
+ * */
+template <class Particle>
+class PairwiseVerletNeighborList : public VerletListsCellsNeighborListInterface<Particle> {
+ public:
+  /**
+   * Type of the data structure used to save the neighbor lists.
+   * */
+  using listType = typename VerletListsCellsHelpers<Particle>::PairwiseNeighborListsType;
+  /**
+   * Constructor for PairwiseVerletNeighborList. Initializes private attributes.
+   * */
+  PairwiseVerletNeighborList() : _aosNeighborList{}, _particleToCellMap{}, _globalToLocalIndex{} {}
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::getContainerType()
+   * */
+  [[nodiscard]] ContainerOption getContainerType() const override { return ContainerOption::pairwiseVerletLists; }
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::getNumberOfPartners()
+   * */
+  const size_t getNumberOfPartners(const Particle *particle) const override {
+    size_t localSize = 0;
+    const auto [firstCellIndex, particleInCellIndex] = _particleToCellMap.at(const_cast<Particle *>(particle));
+    size_t numberOfCellsToInteract = 27;
+    for (size_t secondCellIndex = 0; secondCellIndex < numberOfCellsToInteract; secondCellIndex++) {
+      localSize += _aosNeighborList[firstCellIndex][secondCellIndex][particleInCellIndex].second.size();
+    }
+    return localSize;
+  }
+
+  /**
+   * Returns the neighbor list in AoS layout.
+   * @return Neighbor list in AoS layout.
+   * */
+  typename VerletListsCellsHelpers<Particle>::PairwiseNeighborListsType &getAoSNeighborList() {
+    return _aosNeighborList;
+  }
+
+  /**
+   * @copydoc VerletListsCellsNeighborListInterface::buildAoSNeighborList()
+   * */
+  void buildAoSNeighborList(LinkedCells<Particle> &linkedCells, bool useNewton3, double cutoff, double skin,
+                            double interactionLength, const TraversalOption buildTraversalOption) override {
+    _aosNeighborList.clear();
+    auto &cells = linkedCells.getCells();
+    auto cellsSize = cells.size();
+    _aosNeighborList.resize(cellsSize);
+    _globalToLocalIndex.resize(cellsSize);
+
+    for (size_t firstCellIndex = 0; firstCellIndex < cellsSize; ++firstCellIndex) {
+      size_t numCellsInteracting = 27;  // every cell has max 26 neighbors + interaction with itself
+      _aosNeighborList[firstCellIndex].resize(numCellsInteracting);
+      for (size_t secondCellIndex = 0; secondCellIndex < numCellsInteracting; ++secondCellIndex) {
+        // reserve vector of neighbor lists for every particle in cell1
+        _aosNeighborList[firstCellIndex][secondCellIndex].reserve(cells[firstCellIndex].numParticles());
+        size_t particleIndexCurrentCell = 0;
+        for (auto particleIter = cells[firstCellIndex].begin(); particleIter.isValid(); ++particleIter) {
```

Comment:
Make all these loops range based. On the one hand side, it will be easier to read, on the other there are not necessarily always be 27 neighbor cells. You probably need to calculate the actual number by fetching the cell length  from the linkedCells container and comparing it to the interaction length.

Reply:
Feel free to make the corresponding loops in VerletListsCellsNeighborList.h range based too.